drivers/ec_host_cmd_periph/ec_host_cmd_npcx.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
```

Comment:
This should be an eSPI driver that works for all eSPI implementations (versus a npcx implementation). NPCX should have an eSPI implementation, but the host cmd periph should be based on a bus like eSPI, SPI, or I2C.

I also believe that I need to update the ec_host_cmd_periph API to include a blocking `read` API function. I used semaphores in the initial API instead, but now that I am implementing the `SPI` version of this periph driver, I realize it make things more complicated than it needs to be.

I have a *very* rough draft of the `SPI` implementation for this driver that I [uploaded](https://github.com/jettr/zephyr/tree/spi_periph_drat) to show you what I mean.

Reply:
That's my original plan, too. eSPI driver offers  espi_api_lpc_read_request, espi_api_lpc_write_request  and callbacks to handle the events from host sub-modules such as KBC, APCI IO ports and so on. The Chromium host cmd framework needs a event from the IO port and handle massive data via shared memory or mailbox. 

But the reality is eSPI api functions only support 8041 so far. Please refer  https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/espi.h#L212 . We need to add more op codes after the interface between host commands and espi peripheral is finalized. In order to test the hardware behavior on npcx7 first, this CL follows what you did fo the simulator and passed the sanity test on my side 

Should I drop this commit in this PR? After all host_cmd_periph_xxx api functions are landed, we can create the other PR to support host command framework.

Reply:
Let's drop the host command peripheral for now then and we can add it when eSPI and the host command framework are both in a better state

Reply:
Got it. I will drop it in this PR.

Reply:
Done.